mono/mini/method-to-ir.c

```diff
@@ -9060,11 +9060,13 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 
 			/* Tail prefix / tail call optimization */
 
-			/* FIXME: Enabling TAILC breaks some inlining/stack trace/etc tests */
+			/* FIXME: Enabling TAILC breaks some inlining/stack trace/etc tests.
+				  Inlining and stack traces are not guaranteed however. */
 			/* FIXME: runtime generic context pointer for jumps? */
 			/* FIXME: handle this for generic sharing eventually */
-			if ((ins_flag & MONO_INST_TAILCALL) &&
-				!vtable_arg && !cfg->gshared && is_supported_tail_call (cfg, method, cmethod, fsig, call_opcode))
+			if ((ins_flag & MONO_INST_TAILCALL)
+					&& (MONO_ARCH_HAVE_OP_TAIL_CALL || (!vtable_arg && !cfg->gshared))
```

Comment:
This code is different than https://github.com/mono/mono/pull/7426/files#diff-e7526071878d708680e4a8c5fa5a46edR9052. Any reason for the change of approach?

Reply:
Yes very slight change.
non-preprocessor dependency, granted the CAPS could be defined later to be backend->lower instead of being effectively a  pound-if, if/when we later have multiple backends

Reply:
We can skip this PR & just take the other if you want. It should also be possible to enable some tests now for some targets (x86 amd64 both arms).